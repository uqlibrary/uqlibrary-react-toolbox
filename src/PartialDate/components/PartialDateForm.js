import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';

const moment = require('moment');
class PartialDateForm extends Component {
    static propTypes = {
        locale: PropTypes.object,
        onChange: PropTypes.func,
        dateFormat: PropTypes.string,
        allowPartial: PropTypes.bool,
        months: PropTypes.array
    };

    static defaultProps = {
        locale: {
            dayLabel: 'Day',
            monthLabel: 'Month',
            yearLabel: 'Year',
            validationMessage: {
                day: 'Invalid day',
                month: 'Invalid month',
                year: 'Invalid year'
            },
            minNumberCharCode: 48,
            maxNumberCharCode: 57
        },
        months: [ 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December' ],
        dateFormat: 'YYYY-MM-DD',
        allowPartial: false
    };

    constructor(props) {
        super(props);
        this.state = {
            day: null,
            month: null,
            year: null
        };
        this.errors = {};
    }

    componentWillUpdate(nextProps, nextState) {
        if (this.props.onChange) this.props.onChange(this._setDate(nextState));
    }

    _validate = (state) => {
        let valid;
        const { day, month, year } = state;
        const { locale } = this.props;

        this.errors.year = isNaN(year) ? locale.validationMessage.year : '';

        if (this.props.allowPartial) {
            valid = !isNaN(year) && year !== null && moment(state).isValid();
            this.errors.month = (year && month < 0) ? locale.validationMessage.month : '';
            this.errors.day = (day && year && month > -1 && !valid) ? locale.validationMessage.day : '';
        } else {
            valid = moment(state).isValid() && !isNaN(day) && day !== null && !isNaN(year) && year !== null && month !== null;
            this.errors.month = month < 0  ? locale.validationMessage.month : '';
            this.errors.day = (isNaN(day) || ((month !== null || month > -1) && year && !valid))  ? locale.validationMessage.day : '';
        }

        return valid;
    };

    _setDate = (date) => {
        if (this._validate(date)) {
            if (this.props.allowPartial) {
                date.month = date.month < 0 ? 0 : date.month;
            }
            return moment(date).format(this.props.dateFormat);
        }
        return '';
    };

    _isNumber = (event) => ((event.charCode < this.props.locale.minNumberCharCode || event.charCode > this.props.locale.maxNumberCharCode) && event.preventDefault());

    _onDateChanged = (key) => (event, index, value) => this.setState({ [key]: parseInt(event.target.value === undefined ? value : event.target.value, 10)});

    render() {
        const { locale, months } = this.props;
        const renderMonths = months.map((month, index) =>
            <MenuItem
                key={index}
                value={index}
                primaryText={month}
            />
        );

        return (
            <div className="column">
                <div className="columns">
                    <div className="column">
                        <TextField
                            name="day"
                            type="text"
                            maxLength="2"
                            style={{ marginTop: '12px' }}
                            fullWidth
                            floatingLabelText={ locale.dayLabel }
                            floatingLabelFixed
                            errorText={ this.errors.day }
                            onKeyPress={ this._isNumber }
                            onChange={ this._onDateChanged('day') }
                            onBlur={ !this.props.allowPartial ? this._onDateChanged('day') : undefined }
                        />
                    </div>
                    <div className="form-spacer"/>
                    <div className="column">
                        <SelectField
                            name="month"
                            fullWidth
                            value={ this.state.month }
                            style={{ marginTop: '12px' }}
                            floatingLabelText={ locale.monthLabel }
                            floatingLabelFixed
                            errorText={ this.errors.month }
                            onChange={ this._onDateChanged('month') }>
                            <MenuItem key={-1} value={-1} primaryText=""/>
                            { renderMonths }
                        </SelectField>
                    </div>
                    <div className="form-spacer"/>
                    <div className="column">
                        <TextField
                            name="year"
                            type="text"
                            fullWidth
                            style={{ marginTop: '12px' }}
                            maxLength="4"
                            floatingLabelText={ locale.yearLabel }
                            floatingLabelFixed
                            errorText={ this.errors.year }
                            onKeyPress={ this._isNumber }
                            onChange={ this._onDateChanged('year') }
                            onBlur={ this._onDateChanged('year') }
                        />
                    </div>
                </div>
            </div>
        );
    }
}


export default PartialDateForm;
