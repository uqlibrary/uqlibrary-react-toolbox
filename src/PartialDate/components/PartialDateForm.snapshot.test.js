jest.dontMock('./PartialDateForm');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import PartialDateForm from './PartialDateForm';

function setup(props) {
    return shallow(<PartialDateForm {...props} />);
}

describe('PartialDateForm snapshots tests', () => {
    it('renders PartialDateForm component', () => {
        const props = {
            name: 'partialDate',
            allowPartial: true,
            onChange: () => {}
        };

        const wrapper = setup(props);

        let tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();

        wrapper.find('year').simulate('blur');
    });
});
