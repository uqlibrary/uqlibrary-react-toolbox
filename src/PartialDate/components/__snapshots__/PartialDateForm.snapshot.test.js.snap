// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartialDateForm snapshots tests renders PartialDateForm component 1`] = `
<div
  className="columns is-gapless-mobile"
>
  <div
    className="column"
  >
    <TextField
      className=""
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Day"
      fullWidth={true}
      maxLength="2"
      multiLine={false}
      name="day"
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
  <div
    className="column"
  >
    <SelectField
      autoWidth={false}
      className=""
      disabled={false}
      dropDownMenuProps={
        Object {
          "animated": false,
        }
      }
      floatingLabelFixed={true}
      floatingLabelText="Month"
      fullWidth={true}
      multiple={false}
      name="month"
      onChange={[Function]}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      value={null}
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="-1"
        primaryText=""
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={-1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="0"
        primaryText="January"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={0}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="1"
        primaryText="February"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="2"
        primaryText="March"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={2}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="3"
        primaryText="April"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={3}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="4"
        primaryText="May"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={4}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="5"
        primaryText="June"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={5}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="6"
        primaryText="July"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={6}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="7"
        primaryText="August"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={7}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="8"
        primaryText="September"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={8}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="9"
        primaryText="October"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={9}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="10"
        primaryText="November"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={10}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="11"
        primaryText="December"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={11}
      />
    </SelectField>
  </div>
  <div
    className="column"
  >
    <TextField
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Year"
      fullWidth={true}
      maxLength="4"
      multiLine={false}
      name="year"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
</div>
`;

exports[`PartialDateForm snapshots tests renders PartialDateForm component 2 1`] = `
<div
  className="columns is-gapless-mobile"
>
  <div
    className="column"
  >
    <TextField
      className=""
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Day"
      fullWidth={true}
      maxLength="2"
      multiLine={false}
      name="day"
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
  <div
    className="column"
  >
    <SelectField
      autoWidth={false}
      className=""
      disabled={false}
      dropDownMenuProps={
        Object {
          "animated": false,
        }
      }
      floatingLabelFixed={true}
      floatingLabelText="Month"
      fullWidth={true}
      multiple={false}
      name="month"
      onChange={[Function]}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      value={null}
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="-1"
        primaryText=""
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={-1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="0"
        primaryText="January"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={0}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="1"
        primaryText="February"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="2"
        primaryText="March"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={2}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="3"
        primaryText="April"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={3}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="4"
        primaryText="May"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={4}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="5"
        primaryText="June"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={5}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="6"
        primaryText="July"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={6}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="7"
        primaryText="August"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={7}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="8"
        primaryText="September"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={8}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="9"
        primaryText="October"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={9}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="10"
        primaryText="November"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={10}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="11"
        primaryText="December"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={11}
      />
    </SelectField>
  </div>
  <div
    className="column"
  >
    <TextField
      className="requiredField"
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Year"
      fullWidth={true}
      maxLength="4"
      multiLine={false}
      name="year"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
</div>
`;

exports[`PartialDateForm snapshots tests renders PartialDateForm component 3 1`] = `
<div
  className="columns is-gapless-mobile"
>
  <div
    className="column"
  >
    <TextField
      className="requiredField"
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Day"
      fullWidth={true}
      maxLength="2"
      multiLine={false}
      name="day"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
  <div
    className="column"
  >
    <SelectField
      autoWidth={false}
      className="requiredField"
      disabled={false}
      dropDownMenuProps={
        Object {
          "animated": false,
        }
      }
      floatingLabelFixed={true}
      floatingLabelText="Month"
      fullWidth={true}
      multiple={false}
      name="month"
      onChange={[Function]}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      value={null}
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="-1"
        primaryText=""
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={-1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="0"
        primaryText="January"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={0}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="1"
        primaryText="February"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={1}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="2"
        primaryText="March"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={2}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="3"
        primaryText="April"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={3}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="4"
        primaryText="May"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={4}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="5"
        primaryText="June"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={5}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="6"
        primaryText="July"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={6}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="7"
        primaryText="August"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={7}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="8"
        primaryText="September"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={8}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="9"
        primaryText="October"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={9}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="10"
        primaryText="November"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={10}
      />
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        key="11"
        primaryText="December"
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
        value={11}
      />
    </SelectField>
  </div>
  <div
    className="column"
  >
    <TextField
      className="requiredField"
      disabled={false}
      floatingLabelFixed={true}
      floatingLabelText="Year"
      fullWidth={true}
      maxLength="4"
      multiLine={false}
      name="year"
      onBlur={[Function]}
      onChange={[Function]}
      onKeyPress={[Function]}
      rows={1}
      style={
        Object {
          "marginTop": "12px",
        }
      }
      type="text"
      underlineShow={true}
    />
  </div>
</div>
`;
